app/mutations/BaseMutation.ts

```diff
@@ -53,10 +53,25 @@ export default class BaseMutation<T extends BaseMutationType = never> {
       }
     ) {
       return new Promise<T['response']>((resolve, reject) => {
-        commitMutationDefault<T>(environment, {
+        // Debounced mutations should be commited immediately to perform the optimisticUpdate
+        // The actual request will be cancelled in the network layer
+        // Here we either dispose of a debounced mutation, or remove it from the map when it errors/completes
+        if (debounceKey) {
+          const previousMutation = debouncedMutationMap.get(debounceKey);
+          if (previousMutation) {
+            previousMutation.dispose();
```

Comment:
perfect! We talked about some hiccups with disposables not getting fully disposed...did we resolve the edge case of disposing network-requests that are already in flight?

Reply:
Right. If the mutation made it past the `debounceMiddleware` then we may dispose of a mutation that will make its way to the serve. I guess the milddeware should remove the mutation from the `debouncedMutationMap` before calling `next()`. Will update that